configuration/network.go

```diff
@@ -13,47 +13,20 @@
 // Arduino software without disclosing the source code of your own applications.
 // To purchase a commercial license, send an email to license@arduino.cc.
 
-package httpclient
+package configuration
 
 import (
 	"fmt"
 	"net/url"
 	"runtime"
 
 	"github.com/arduino/arduino-cli/cli/globals"
-	"github.com/arduino/arduino-cli/configuration"
+	"github.com/spf13/viper"
 )
 
-// Config is the configuration of the http client
-type Config struct {
-	UserAgent string
-	Proxy     *url.URL
-}
-
-// DefaultConfig returns the default http client config
-func DefaultConfig() (*Config, error) {
-	var proxy *url.URL
-	var err error
-	if configuration.Settings.IsSet("network.proxy") {
-		proxyConfig := configuration.Settings.GetString("network.proxy")
-		if proxyConfig == "" {
-			// empty configuration
-			// this workaround must be here until viper can UnSet properties:
-			// https://github.com/spf13/viper/pull/519
-		} else if proxy, err = url.Parse(proxyConfig); err != nil {
-			return nil, fmt.Errorf(tr("Invalid network.proxy '%[1]s': %[2]s"), proxyConfig, err)
-		}
-	}
-
-	return &Config{
-		UserAgent: UserAgent(),
-		Proxy:     proxy,
-	}, nil
-}
-
-// UserAgent returns the user agent for the cli http client
-func UserAgent() string {
-	subComponent := configuration.Settings.GetString("network.user_agent_ext")
+// UserAgent returns the user agent for the cli (mainly used by http clients)
```

Comment:
Uhm actually the wording `for the cli` is redundant, since this is the `configuration` package that is used in the Arduino CLI project there is no need to further specify it.